vignettes/concepts_conventions.Rmd

```diff
@@ -0,0 +1,288 @@
+to achieve this because expressions inherently pause evaluation of your code before it's executed:
+
+```{r, eval = TRUE}
+a <- expr(2)
+b <- expr(3)
+
+expr(a + b)
+# NOT 2 + 3
+```
+
+This is where `!!` (bang-bang) comes in: provided again by the `{rlang}` package, it  allows you to inject the contents of an 
+expression into another expression, meaning that by using `!!` you can modify the code inside an expression before R evaluates it.
+By using `!!` you are **unquoting** an expression, i.e. evaluating it before you pass it onwards.
+
+
+```{r, eval = TRUE}
+expr(!!a + !!b)
+```
+
+You can see an example of where `!!` comes in handy within `{admiral}` code in [Common Pitfall 1](#pitfall1), where the contents of an expression is unquoted so that it can be passed to `derive_vars_merged()`.
+
+`!!!` (bang-bang-bang) is the plural version of `!!` and can be used to unquote a list of expressions:
+
+```{r, eval = TRUE}
+exprs(!!!list(a, b))
+```
+
+Within `{admiral}`, this operator can be useful if we need to unquote a list of variables (stored as expressions) to use them inside
+of an `{admiral}` or even `{dplyr}` call. One example is the `{admiral}` subject keys:
+
+```{r, eval = TRUE}
+get_admiral_option("subject_keys")
+```
+
+If we want to use the subject keys stored within this `{admiral}` option to subset a dataset, we need to use `!!!` to unquote this list.
+Let's construct a dummy example to illustrate the point:
+
+```{r, eval = TRUE, error = TRUE}
+adcm <- data.frame(STUDYID = "XXX", USUBJID = "XXX-1", CMTRT = "ASPIRIN")
+adcm
+
+# This doesn't work as we are not unquoting the subject keys
+adcm %>% select(get_admiral_option("subject_keys"))
+
+# This works because we are unquoting the subject keys
+adcm %>% select(!!!get_admiral_option("subject_keys"))
+```
+
+You can see another example of `!!!` in action in [this line](https://github.com/pharmaverse/admiral/blob/c5dd8fb196c2cd3f27a6b02ec88fa1abd6b24d60/inst/templates/ad_adex.R#L162) of the
+`{admiral}` `ADEX` template script, where it is used to dynamically control the by variables passed to an `{admiral}` function.
+
+### Summary
+
+In summary, although the expressions framework may seem slightly clunky and mysterious to begin with, it allows for such power
+and flexibility that it forms a key part of the `{admiral}` package. For a comprehensive treatment of expressions, see 
+[Chapter 18](https://adv-r.hadley.nz/expressions.html) and [Chapter 19](https://adv-r.hadley.nz/quasiquotation.html) of the Advanced R textbook. Chapter 19 specifically covers in much more detail the concept of unquoting.
+
+## Common pitfalls 
+
+Expressions are very powerful, but this can also lead to misunderstandings about their functionality. Let's set up some dummy
+data to explore common issues that new (or experienced!) programmers may encounter when dealing with expressions.
```

Comment:
sneaky little hobbit's comment: are only programmers using {admiral} ? or could it be more general with "users" instead?

Reply:
i think if you are using admiral you are programming, so happy to keep 😄